From fd49c47dee908ef46dda29184c719009fe09c62a Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Tue, 1 Sep 2015 19:11:50 +0800
Subject: [PATCH 0386/5242] MLK-11480-03 ARM: dts: add spdif sound card
 support for imx6sx-sdb board

commit  b7022d269840722c6feb4b8da0a30fcb95fd7a01 from
https://source.codeaurora.org/external/imx/linux-imx.git

add spdif sound card support for imx6sx-sdb board

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sx-sdb.dtsi |   20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm/boot/dts/imx6sx-sdb.dtsi b/arch/arm/boot/dts/imx6sx-sdb.dtsi
index 9a80eb0..ede9dd4 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dtsi
+++ b/arch/arm/boot/dts/imx6sx-sdb.dtsi
@@ -161,6 +161,14 @@
 		hp-det-gpios = <&gpio1 17 1>;
 	};
 
+	sound-spdif {
+		compatible = "fsl,imx-audio-spdif",
+			   "fsl,imx6sx-sdb-spdif";
+		model = "imx-spdif";
+		spdif-controller = <&spdif>;
+		spdif-out;
+	};
+
 	panel {
 		compatible = "sii,43wvf1g";
 		backlight = <&backlight_display>;
@@ -313,6 +321,12 @@
 	status = "disabled";
 };
 
+&spdif {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spdif>;
+	status = "okay";
+};
+
 &ssi2 {
 	status = "okay";
 };
@@ -563,6 +577,12 @@
 			>;
 		};
 
+		pinctrl_spdif: spdifgrp {
+			fsl,pins = <
+				MX6SX_PAD_SD4_DATA4__SPDIF_OUT          0x1b0b0
+			>;
+		};
+
 		pinctrl_vcc_sd3: vccsd3grp {
 			fsl,pins = <
 				MX6SX_PAD_KEY_COL1__GPIO2_IO_11		0x17059
-- 
1.7.9.5

